src/store/global/global.thunks.ts

```diff
@@ -834,11 +840,16 @@ function fetchCoordinatorState() {
  */
 function disconnectWallet() {
   return (dispatch: AppDispatch) => {
-    const provider = Providers.getProvider();
-    if (provider.provider?.connector) {
-      // Kills the stored Web Connect session to show QR with next login
-      provider.provider.connector.killSession();
-    }
+    // const provider = Providers.getProvider();
+
+    // ToDo: This does not seem to exist in the ethers.providers.Web3Provider
+    //       Is the type wrong or the implementation outdated?
+
+    // if (provider.provider?.connector) {
+    //   // Kills the stored Web Connect session to show QR with next login
+    //   provider.provider.connector.killSession();
```

Comment:
An `ethers` provider is just an interface that could be extended. In this case, if the provider is of type `WalletConnectProvider`, which is imported from the `@walletconnect/web3-provider` package we need to call this to kill the `WalletConnect` session. Ping me in Kyebase if you prefer it and I'll provide more info!

Reply:
It's clear.

We now test whether the provider implements the interface [IConnector](https://github.com/WalletConnect/walletconnect-monorepo/blob/v1.0/packages/helpers/types/index.d.ts#L2) to be able to call `killSession`: d4a65e1a98165c7d4ee12ad38dc735c96a890fea.

Reply:
Amazing! :)